gulpfile.js

```diff
@@ -53,63 +54,89 @@ gulp.task('clean/dotnet', function () {
         .pipe(clean({ force: true }));
 });
 
-gulp.task('restore/dotnet', ['clean/dotnet'], function () {
+gulp.task('restore/dotnet', gulp.series(['clean/dotnet'], function () {
     console.log('Running dotnet restore...');
     return gulp.src('src/dotnet/**/*.csproj')
         .pipe(restore());
-});
+}))
 
-gulp.task('build/dotnet', ['restore/dotnet'], function () {
+gulp.task('build/dotnet', gulp.series(['restore/dotnet'], function () {
     console.log('Running dotnet build...');
     return gulp.src('src/dotnet/**/*.csproj')
         .pipe(build());
-});
+}))
 
-gulp.task('test/dotnet', ['build/dotnet'], function () {
+gulp.task('test/dotnet', gulp.series(['build/dotnet'], function () {
     console.log('Running the dotnet unit tests...');
     return gulp.src('src/dotnet/OpenAPI.Validator.Tests/OpenAPI.Validator.Tests.csproj')
         .pipe(test({ verbosity: 'minimal' }));
-});
+}))
 
 // Now the defaults/commons
-gulp.task('build', ['build/dotnet', 'build/typescript'], function () {
-    console.log('Building code...');
-});
-
-gulp.task('clean', ['clean/typescript', 'clean/dotnet'], function () {
-    console.log('Cleaning artifacts...');
-});
-
-gulp.task('test', ['test/dotnet', 'test/typescript'], function () {
-    console.log('Successfully ran the tests...');
-});
-
-gulp.task('dotnet', ['test/dotnet'], function () {
-});
-
-gulp.task('typescript', ['test/typescript'], function () {
-});
-
-gulp.task('default', ['dotnet', 'typescript'], function () {
-    console.log('Successfully built and tested the repo...');
-});
-
-gulp.task('dotnet/pack', ['dotnet', 'typescript'], function () {
-    console.log('Kicking off the dotnet publish task...');
-    return gulp.src('src/dotnet/AutoRest/AutoRest.csproj')
-        .pipe(run('dotnet publish src/dotnet/AutoRest/AutoRest.csproj --configuration release --output bin/netcoreapp2.0'));
-});
-
-gulp.task('pack/typescript', [], function () {
+gulp.task(
+  "build",
+  gulp.parallel(["build/dotnet", "build/typescript"], function (done) {
+    console.log("Building code...")
+    done()
+  })
+)
+
+gulp.task(
```

Comment:
Why you add gulp.parallel and gulp.series? what do they do?

Reply:
The two fucntions is required in  new version of gulp, they determin the task will  run series or parallel